packages/chain-adapters/src/utxo/UTXOChainAdapter.ts

```diff
@@ -40,30 +40,35 @@ export interface ChainAdapterArgs {
     ws: bitcoin.ws.Client
   }
   coinName: string
+  coinType: ChainTypes
+  curve?: string
+  bip44Params: BIP44Params
+
 }
 
 export class ChainAdapter implements IChainAdapter<ChainTypes.Bitcoin> {
   private readonly providers: {
     http: bitcoin.api.V1Api
     ws: bitcoin.ws.Client
   }
-
-  public static readonly defaultBIP44Params: BIP44Params = {
-    purpose: 84, // segwit native
-    coinType: 0,
-    accountNumber: 0
-  }
-
-  // TODO(0xdef1cafe): constraint this to utxo coins and refactor this to be a UTXOChainAdapter
-  coinName: string
+  private readonly coinName: string
+  private readonly coinType: ChainTypes
+  private readonly bip44Params: BIP44Params
+  private readonly curve: string = 'secp256k1'
 
   constructor(args: ChainAdapterArgs) {
```

Comment:
these args will need to be generic too, as in
```
export interface ChainAdapterArgs<T extends ChainTypes> {
  // ...
  chainType: T
  // ...
}
```

```suggestion
  constructor(args: ChainAdapterArgs<T>) {
```